internal/instrument/instrument.go

```diff
@@ -965,23 +983,60 @@ func ginMiddleware(got *dst.AssignStmt) dst.Stmt {
 	if !ok {
 		return nil
 	}
+	stmt := useMiddleware(iden.Name, "GinMiddleware")
+	wrap(stmt)
+	return stmt
+}
+
+func isEcho(stmt *dst.AssignStmt) bool {
+	rhs := stmt.Rhs[0]
+	f, ok := funcIdent(rhs)
+	return ok && f.Path == "github.com/labstack/echo/v4" && f.Name == "New"
+}
+
+func echoMiddleware(got *dst.AssignStmt) dst.Stmt {
+	iden, ok := got.Lhs[0].(*dst.Ident)
+	if !ok {
+		return nil
+	}
+	stmt := useMiddleware(iden.Name, "EchoMiddleware")
+	wrap(stmt)
+	return stmt
+}
```

Comment:
Food for thought: I feel it would make sense to separate these specific framework methods in their own files. WDYT?

Reply:
100% agree! The file is becoming huge. Feel free to do it in a separate PR though + let's also do it for all contribs not only echo.